api/v1beta1/agentserviceconfig_types.go

```diff
@@ -88,6 +88,12 @@ type AgentServiceConfigSpec struct {
 	// that are used if one the operators fails to be successfully deployed
 	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Must-Gather Images"
 	MustGatherImages []MustGatherImage `json:"mustGatherImages,omitempty"`
+
+	// ExposeIPXEHTTPRoute is controlling whether the operator is creating plain HTTP routes
+	// iPXE hosts may not work with router cyphers and may access artifacts via HTTP only
+	// This setting defaults to false
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Expose IPXE HTTP route"
+	ExposeIPXEHTTPRoute bool `json:"exposeIPXEHTTPRoute,omitempty"`
```

Comment:
Even if we make it very targeted I think we should still try for a string here.

Maybe `HTTPIPXERoute` with `enabled` or `disabled` as possibly values?
Does that really gain us anything over a bool though? I don't see a whole lot of room for expansion.

@mhrivnak @avishayt do you want to put some suggestions here?

Reply:
That'd make us parse the string instead of golang's json.Marshal, I don't see any benefit tbh

Reply:
> That'd make us parse the string instead of golang's json.Marshal, I don't see any benefit tbh

Not sure I understand this. The string key and value will get into the structure just fine from json.
The motivation is the same that we referenced in slack which is the [kubernetes API conventions around primitive types](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types)

Reply:
Ah, okay, I guess `Enabled` could be a good choice for a start